errai-common-apt/src/main/java/org/jboss/errai/common/apt/generator/AbstractErraiModuleExportFileGenerator.java

```diff
@@ -0,0 +1,113 @@
+import org.jboss.errai.common.apt.AptAnnotatedElementsFinder;
+import org.jboss.errai.common.apt.exportfile.ExportFile;
+import org.jboss.errai.common.apt.exportfile.ExportFileName;
+
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.RoundEnvironment;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.TypeElement;
+import javax.tools.JavaFileObject;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.Set;
+
+import static java.util.stream.Collectors.toSet;
+import static org.jboss.errai.common.apt.ErraiAptPackages.exportFilesPackagePath;
+
+/**
+ * @author Tiago Bento <tfernand@redhat.com>
+ */
+public abstract class AbstractErraiModuleExportFileGenerator extends AbstractProcessor {
+
+  @Override
+  public boolean process(final Set<? extends TypeElement> annotations, final RoundEnvironment roundEnv) {
+
+    try {
+      APTClassUtil.init(processingEnv.getTypeUtils(), processingEnv.getElementUtils());
+      generateAndSaveExportFiles(annotations, new AptAnnotatedElementsFinder(roundEnv));
+    } catch (final Exception e) {
+      System.out.println("Error generating export files");
+      e.printStackTrace();
+    }
+
+    return false;
+  }
+
+  void generateAndSaveExportFiles(final Set<? extends TypeElement> annotations,
+          final AnnotatedElementsFinder annotatedElementsFinder) {
+
+    buildExportFiles(annotations, annotatedElementsFinder).forEach(this::generateSourceAndSave);
+  }
+
+  Set<ExportFile> buildExportFiles(final Set<? extends TypeElement> annotations,
+          final AnnotatedElementsFinder annotatedElementsFinder) {
+
+    return annotations.stream()
+            .map(annotation -> new ExportFile(getModuleName(), annotation,
+                    annotatedClassesAndInterfaces(annotatedElementsFinder, annotation)))
+            .filter(ExportFile::hasExportedTypes)
+            .collect(toSet());
+  }
+
+  Set<? extends Element> annotatedClassesAndInterfaces(final AnnotatedElementsFinder annotatedElementsFinder,
+          final TypeElement annotationTypeElement) {
+
+    return annotatedElementsFinder.getElementsAnnotatedWith(annotationTypeElement)
+            .stream()
+            .filter(e -> e.getKind().isClass() || e.getKind().isInterface())
+            .collect(toSet());
+  }
+
+  private void generateSourceAndSave(final ExportFile exportFile) {
+
+    final Set<APTClass> exportedTypes = exportFile.exportedTypes.stream()
+            .map(s -> new APTClass(s.asType()))
+            .collect(toSet());
+
+    final String className = ExportFileName.encodeAnnotationNameAsExportFileName(exportFile);
+    final String fullClassName = exportFilesPackagePath() + "." + className;
+    final ClassStructureBuilder<?> classBuilder = ClassBuilder.define(fullClassName).publicScope().body();
+
+    exportedTypes.forEach(
+            exportedType -> classBuilder.publicField(RandomStringUtils.randomAlphabetic(6), exportedType).finish());
```

Comment:
So I think I understand why you are using random Strings here. Personally, my preference is to, wherever possible, use deterministic, stable, and human-readable identifiers.

This isn't a blocker for merging this PR, but I would like to improve on this in the future.

Reply:
Agree 100%. That shouldn't be hard to change, I'll se what I can do.